app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -16,17 +22,28 @@ public class Parse {
     public HashMap<String, Contents> constRepo = new HashMap<>();
     public HashMap<String, ArrayList<StructContents>> structRepo = new HashMap<>();
 
-    public Parse(String code) throws Exception {
-        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+    public static Optional<Parse> parseTextFile(String fileName) throws Exception {
```

Comment:
Similarly, I'd probably just make this throw an exception if the code file failed to open. If you change OpenTextFile to throw an exception, you can just call it directly and assume success (and if not successful, OpenTextFile will throw the exception for you).